pandas/core/generic.py

```diff
@@ -698,6 +702,10 @@ def pop(self: FrameOrSeries, item) -> FrameOrSeries:
         2    lion       80.5
         3  monkey        NaN
         """
+    )
+    @Substitution(klass="DataFrame")
```

Comment:
You would need to import
```python
if TYPE_CHECKING:
    from pandas.core.series import Series  # noqa: F401
```
and then, in the signature, write it as
```python
def pop(self, item: Label) -> Union["Series", Any]:
```

As Simon said, `self` doesn't need an annotation

Reply:
> You would need to import
> 
> ```python
> if TYPE_CHECKING:
>     from pandas.core.series import Series  # noqa: F401
> ```

I don't think i would need this import as you mentioned to use "Series"

> ``` def pop(self, item: Label) -> Union["Series", Any]: ```

Also Flake8 was also suggesting these

`` def pop(self, item: Label) -> Union["Series", Any]: ``